backend/src/bundles/bsa-badges/bsa-badges.repository.ts

```diff
@@ -0,0 +1,35 @@
+import { BSABadgeEntity } from '~/bundles/bsa-badges/bsa-badges.entity.js';
+import { type BSABadgesModel } from '~/bundles/bsa-badges/bsa-badges.model.js';
+import { ErrorMessages } from '~/common/enums/enums.js';
+import { type Repository } from '~/common/types/repository.type.js';
+
+class BSABadgesRepository implements Repository {
+    private bsaBadgesModel: typeof BSABadgesModel;
+
+    public constructor(bsaBadgesModel: typeof BSABadgesModel) {
+        this.bsaBadgesModel = bsaBadgesModel;
+    }
+
+    public async findAll(): Promise<BSABadgeEntity[]> {
+        const badges = await this.bsaBadgesModel.query().execute();
+        return badges.map((it) => BSABadgeEntity.initialize(it));
```

Comment:
You can use just:
```suggestion
        return badges.map(BSABadgeEntity.initialize);
```

Reply:
Do not pass function directly to `.map(…)`. eslint 
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/92392661/2e9877ba-0faf-4ddb-9291-0ebd900103d5)